### PR TITLE
deps: add Scalapack

### PR DESCRIPTION
**Pull request by scivision**
Scalapack uses a library stack that can be challenging to manage.
Not least of all since many Scalapacks ship with broken / incomplete
pkg-config files and CMake FindScalapack.cmake

This resolves those issues for typical Scalapack setups including:

* Linux: Intel MKL or OpenMPI + Netlib
* MacOS: Intel MKL or OpenMPI + Netlib
* Windows: Intel MKL (OpenMPI not available on Windows)

### CODE REVIEW

File: .github/workflows/ci_frameworks.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - run: python -m pip install .
+    - name: install prereq
+      run: |
+        sudo apt update -yq
+        sudo apt install -yq --no-install-recommends pkg-config ninja-build gfortran liblapack-dev libscalapack-mpi-dev libopenmpi-dev openmpi-bin
```
Comment (jpakkane) [inline .github/workflows/ci_frameworks.yml]: This needs to be in the regular travis CI image. All Linux dep tests must be in the same place. Does it require any other additional packages than `libscalapack-mpi-dev`?

Comment (scivision) [reply .github/workflows/ci_frameworks.yml]: OK--right only thing new is `liblapack-dev libscalapack-mpi-dev`

Comment (scivision) [reply .github/workflows/ci_frameworks.yml]: so those would have to be added to Docker or?

Comment (scivision) [reply .github/workflows/ci_frameworks.yml]: For example, HDF5 is in Docker, but I added a GitHub Action to do MacOS as well. When MSYS2 is ready for GitHub Action, I could add that for HDF5 also.

For Scalapack, if I'm understanding correctly, we would add Scalapack to Docker and then perhaps have this GitHub Action for Scalapack on MacOS? 

For many other prereqs, we can also do Windows--OpenMPI is generally not available for Windows so I didn't make an Action for Scalapack on Windows.

Comment (jpakkane) [reply .github/workflows/ci_frameworks.yml]: Due to access permissions I have to update the image manually. I have started the image build, I'll post an update once it has finished uploading so you can restart the build.

However do update the `Dockerfile` to have the new packages.

Comment (jpakkane) [reply .github/workflows/ci_frameworks.yml]: Docker image updated.

File: mesonbuild/dependencies/__init__.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 from .dev import GMockDependency, GTestDependency, LLVMDependency, ValgrindDependency
 from .coarrays import CoarrayDependency
 from .mpi import MPIDependency
-from .misc import (BlocksDependency, HDF5Dependency, NetCDFDependency, OpenMPDependency, Python3Dependency, ThreadDependency, PcapDependency, CupsDependency, LibWmfDependency, LibGCryptDependency, GpgmeDependency, ShadercDependency)
+from .scalapack import ScalapackDependency
+from .misc import (BlocksDependency, HDF5Dependency, MPIDependency, NetCDFDependency, OpenMPDependency, Python3Dependency, ThreadDependency, PcapDependency, CupsDependency, LibWmfDependency, LibGCryptDependency, GpgmeDependency, ShadercDependency)
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/__init__.py]: __[Flake8]__ redefinition of unused 'MPIDependency' from line 23 ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6228#issue-4878846))

<details>

| Rule |
| - |
| F811 |

</details>
<sub>posted by Sider</sub>

**Pull request merged**